src/init/variables/VarResolver.ts

```diff
@@ -0,0 +1,104 @@
+const defaultVarConfig = `${modulePathPlaceholder}config/app-setup/vars.json`;
+
+export type YargsArgOptions = Record<string, yargs.Options>;
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function setupYargvWithBaseArgs(yargv: yargs.Argv) {
+  // Values from componentsjs LogLevel type
+  const logLevels = [ 'error', 'warn', 'info', 'verbose', 'debug', 'silly' ];
+  return yargv.options({
+    /**
+   * CLI options needed for performing meta-process of app initialization.
+   * These options doesn't contribute to components-js vars normally.
+   */
+    config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
+    loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true, choices: logLevels },
+    mainModulePath: { type: 'string', alias: 'm', requiresArg: true },
+    varConfig: { type: 'string', alias: 'v', default: defaultVarConfig, requiresArg: true },
+  });
+}
+
+export interface YargvOptions {
+  usage?: string;
+  strictMode?: boolean;
+  // @see http://yargs.js.org/docs/#api-reference-envprefix
+  loadFromEnv?: boolean;
+  envVarPrefix?: string;
+}
+
+export type VarRecord = Record<string, unknown>;
+
+export class VarResolver extends AsyncHandler<string[], VarRecord> {
+  protected readonly yargsArgOptions: YargsArgOptions;
+  protected readonly yargvOptions: YargvOptions;
+  protected readonly varComputers: Record<string, VarComputer>;
+
+  /**
+   * @param yargsArgOptions - record of option to it's yargs opt config. @range {json}
+   * @param yargvOptions - options to configure yargv. @range {json}
+   * @param varComputers  - record of componentsjs var-iri to VarComputer.
+   */
+  public constructor(
+    yargsArgOptions: YargsArgOptions, yargvOptions: YargvOptions, varComputers: Record<string, VarComputer>,
+  ) {
+    super();
+    this.yargsArgOptions = yargsArgOptions;
+    this.yargvOptions = yargvOptions;
+    this.varComputers = varComputers;
+  }
+
+  private async parseArgs(argv: readonly string[]): Promise<yargs.Arguments> {
+    let yArgv = yargs(argv.slice(2));
+    if (this.yargvOptions.usage !== undefined) {
+      yArgv = yArgv.usage(this.yargvOptions.usage);
+    }
+    if (this.yargvOptions.strictMode) {
+      yArgv = yArgv.strict();
+    }
+    if (this.yargvOptions.loadFromEnv) {
+      yArgv = yArgv.env(this.yargvOptions.envVarPrefix ?? '');
+    }
+
+    yArgv = yArgv.check((args): boolean => {
+      if (args._.length > 0) {
+        throw new Error(`Unsupported positional arguments: "${args._.join('", "')}"`);
+      }
+      for (const [ key, val ] of Object.entries(args)) {
+        // We have no options that allow for arrays
+        if (key !== '_' && Array.isArray(val)) {
+          throw new Error(`Multiple values were provided for: "${key}": "${val.join('", "')}"`);
+        }
+      }
+      return true;
+    }).options(this.yargsArgOptions);
+    setupYargvWithBaseArgs(yArgv);
```

Comment:
Although I wonder if it is still needed; maybe those `baseArgs` should just be in the configuration as well.

Reply:
These base args are there for meta-process. That is, only after interpreting them, we are calling this method. Still they are added, because argv still contains them, and they need to be understood by yargs to support `strict` mode properly along with dynamic options setted up.